dali/test/python/test_torch_pipeline_rnnt.py

```diff
@@ -355,77 +354,149 @@ def test_torch_vs_dali_normalize():
         for normalize_type in ['per_feature', 'all_features']:
             yield _testimpl_torch_vs_dali_normalize, normalize_type, device
 
+def torch_frame_splicing(mel_spec, stacking=1, subsampling=1, device='cpu'):
+    mel_spec = torch.tensor(mel_spec, dtype=torch.float32).unsqueeze(0)
+    if device == 'gpu':
+        mel_spec = mel_spec.cuda()
+    out = stack_subsample_frames(mel_spec, stacking=stacking, subsampling=subsampling)
+    out = out.cpu().numpy().squeeze(0)
+    return out
+
+def dali_frame_splicing(mel_spec_data, stacking, subsampling, device='cpu'):
+    @pipeline_def(batch_size=1, device_id=0, num_threads=3)
+    def log_pipe():
+        x = fn.external_source(lambda: mel_spec_data, device=device, batch=False)
+        # Splice frames implementation needs accessing the data shape on the CPU
+        x_orig_shape = fn.external_source(lambda: np.array(mel_spec_data.shape, dtype=np.int32), device='cpu', batch=False)
+        nfeatures = fn.slice(x_orig_shape, 0, 1, axes=(0,))
+        in_len = fn.slice(x_orig_shape, 1, 1, axes=(0,))
+        if stacking > 1:
+            seq = [x]
+            for n in range(1, stacking):
+                f = fn.slice(x, start=n, shape=in_len, axes=(1,), out_of_bounds_policy='pad', fill_values=0)
+                seq.append(f)
+            x = fn.cat(*seq, axis=0)
+            nfeatures = nfeatures * stacking
+        if subsampling > 1:
+            out_len = (in_len + subsampling - 1) // subsampling
+            m = fn.transforms.scale(scale=[subsampling, 1], center=[0.5, 0])
+            x = fn.reshape(x, rel_shape=[1, 1, -1], layout="HWC")  # Layout required by WarpAffine
+            x = fn.warp_affine(x, matrix=m, size=fn.cat(nfeatures, out_len), interp_type=types.INTERP_NN)
+            x = fn.reshape(x, rel_shape=[1, 1], layout="ft")
```

Comment:
I think this logic should be extracted to a function and used in the full pipeline.

Reply:
Done